### PR TITLE
[TRUS-970] Estates Api for registration.

### PR DESCRIPTION
**Pull request by manishsw**
- estates api for registration.

### CODE REVIEW

File: app/uk/gov/hmrc/trusts/connector/DesConnector.scala [hunk 1]
```diff
@@ -40,8 +40,11 @@ import uk.gov.hmrc.trusts.utils.Constants._
 class DesConnectorImpl @Inject()(http: WSHttp, config: AppConfig) extends DesConnector {
 
   lazy val trustsServiceUrl : String = s"${config.desUrl}/trusts"
+  lazy val estatesServiceUrl : String = s"${config.desUrl}/estates"
+
   lazy val matchEndpoint : String = trustsServiceUrl + "/match"
-  lazy val registrationEndpoint : String = trustsServiceUrl + "/registration"
+  lazy val trustRegistrationEndpoint : String = trustsServiceUrl + "/registration"
+  lazy val registrationEstatesEndpoint : String = estatesServiceUrl + "/registration"
```
Comment (jassalrichy) [inline app/uk/gov/hmrc/trusts/connector/DesConnector.scala]: rename as estateRegistrationEndpoint for consistency

File: app/uk/gov/hmrc/trusts/connector/DesConnector.scala [hunk 2]
```diff
@@ -70,12 +73,20 @@ class DesConnectorImpl @Inject()(http: WSHttp, config: AppConfig) extends DesCon
...
+    val response = http.POST[JsValue, RegistrationResponse](trustRegistrationEndpoint, Json.toJson(registration))
     (implicitly[Writes[JsValue]], RegistrationResponse.httpReads, implicitly[HeaderCarrier](hc),global)
 
     response
   }
 
+  override def registerEstate(registration: EstateRegistration)
+  : Future[RegistrationResponse] = {
+    implicit val hc: HeaderCarrier = HeaderCarrier(extraHeaders = desHeaders)
+    val response = http.POST[JsValue, RegistrationResponse](registrationEstatesEndpoint, Json.toJson(registration))
```
Comment (jassalrichy) [inline app/uk/gov/hmrc/trusts/connector/DesConnector.scala]: see above comment

File: app/uk/gov/hmrc/trusts/connector/DesConnector.scala (hunk 1 above)
Comment (manishsw) [reply app/uk/gov/hmrc/trusts/connector/DesConnector.scala]: @jassalrichy  Done.

File: app/uk/gov/hmrc/trusts/connector/DesConnector.scala (hunk 2 above)
Comment (manishsw) [reply app/uk/gov/hmrc/trusts/connector/DesConnector.scala]: @jassalrichy Done.

**Pull request merged**